adapters/repos/backups/backup_manager.go

```diff
@@ -0,0 +1,223 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package backups
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/semi-technologies/weaviate/adapters/repos/db"
+	"github.com/semi-technologies/weaviate/entities/models"
+	"github.com/semi-technologies/weaviate/entities/schema"
+	"github.com/semi-technologies/weaviate/entities/snapshots"
+	"github.com/semi-technologies/weaviate/usecases/backups"
+	"github.com/semi-technologies/weaviate/usecases/sharding"
+	"github.com/sirupsen/logrus"
+)
+
+type shardingStateFunc func(className string) *sharding.State
+
+type backupManager struct {
```

Comment:
I am wondering what's the reason behind moving the implementation from usecases/ to adapters/

The reason why I am asking is that this deviates from the overall design since the logic behind snapshots is moved outside usecases/

Reply:
because the backup manager imported /adapters/repos/db which was causing an import cycle

Reply:
I see your point.
In master branch the BackupManager depend on the repos/db package which is not correct.  
That's fine for now since I could work on it later as part of a re-factoring task

Reply:
We may need to rethink the current snapshot abstractions if the snapshot logic should be placed in usecases.  The problem is that the backupManager imports `adapters/repos/db.DB`, because 

1. it needs access to verify that the target index exists, and
2. it needs the index to create a new instance of the SnapshotterProvider` interface. Currently a `db.Index` is the only type which implements this interface.

Reply:
As we discussed offline, we can address this dependency issue later down the line, leaving the backupManager in adapters for now